pkg/controller/elasticsearch/services/services.go

```diff
@@ -161,29 +161,31 @@ func NewInternalService(es esv1.Elasticsearch, meta metadata.Metadata) *corev1.S
 
 // NewRemoteClusterService returns the service associated to the remote cluster service for the given cluster.
 func NewRemoteClusterService(es esv1.Elasticsearch, meta metadata.Metadata) *corev1.Service {
-	svc := &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        RemoteClusterServiceName(es.Name),
-			Namespace:   es.Namespace,
-			Labels:      meta.Labels,
-			Annotations: meta.Annotations,
-		},
-		Spec: corev1.ServiceSpec{
-			Type:                     corev1.ServiceTypeClusterIP,
-			Selector:                 label.NewLabels(k8s.ExtractNamespacedName(&es)),
-			PublishNotReadyAddresses: true,
-			ClusterIP:                "None",
-		},
+	nsn := k8s.ExtractNamespacedName(&es)
+	svc := corev1.Service{
+		ObjectMeta: es.Spec.RemoteClusterServer.Service.ObjectMeta,
+		Spec:       es.Spec.RemoteClusterServer.Service.Spec,
 	}
-	selector := label.NewLabels(k8s.ExtractNamespacedName(&es))
+
+	svc.ObjectMeta.Namespace = es.Namespace
+	svc.ObjectMeta.Name = RemoteClusterServiceName(es.Name)
+	// Allow connections to pods that are not yet ready
+	svc.Spec.PublishNotReadyAddresses = true
```

Comment:
Not totally sure why we do `PublishNotReadyAddresses = true` in this SVC, neither why it is headless by default, instead of a normal ClusterIP, but anyway that's not in the scope of this PR.